records_mover/url/base.py

```diff
@@ -48,6 +50,34 @@ def directories_in_directory(self) -> List['BaseDirectoryUrl']:
         "Return entries in this directory."
         raise NotImplementedError()
 
+    def empty(self) -> bool:
+        # TODO document
+        return not self.directories_in_directory()
+
+    @contextmanager
+    def temporary_file(self) -> Iterator['BaseFileUrl']:
+        "Yields a temporary FileUrl in current location"
+        num_chars = 8
+        random_slug = secrets.token_urlsafe(num_chars)
+        temp_loc = self.file_in_this_directory(random_slug)
+        try:
+            yield temp_loc
+        finally:
+            try:
+                temp_loc.delete()
+            except Exception as e:
+                logger.debug(e, exc_info=True)
+
+    def writable(self) -> bool:
+        # TODO document
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker